tools/storyteller/StoryWriter.java

```diff
@@ -1,88 +1,82 @@
 package tools.storyteller;
 
-import com.google.protobuf.Message;
 import com.google.startupos.common.FileUtils;
 import com.google.startupos.common.Logger;
 import java.awt.AWTException;
 import java.awt.Rectangle;
 import java.awt.Robot;
 import java.awt.Toolkit;
 import java.awt.image.BufferedImage;
-import java.io.File;
 import java.io.IOException;
 import java.nio.file.Paths;
 import java.text.DateFormat;
 import java.text.SimpleDateFormat;
+import java.util.ArrayList;
 import java.util.Date;
+import java.util.List;
 import javax.imageio.ImageIO;
 import javax.inject.Inject;
 import javax.inject.Singleton;
 import tools.storyteller.Protos.Config;
-import tools.storyteller.Protos.FileData;
-import tools.storyteller.Protos.ScreenshotMetadata;
+import tools.storyteller.service.Protos.Story;
+import tools.storyteller.service.Protos.StoryItem;
+import tools.storyteller.service.Protos.StoryList;
 
 
 /* Storyteller writer that writes story files. */
 @Singleton
 public class StoryWriter {
   private static final Logger log = Logger.getForClass();
-  public static DateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss_z");
+  private static final DateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy-MM-dd_HH:mm:ss_z");
 
-  private Config config;
-  private FileUtils fileUtils;
+  private final Config config;
+  private final FileUtils fileUtils;
+  private Story.Builder currentStoryBuilder;
+  private List<Story> allStories;
 
   @Inject
   public StoryWriter(StorytellerConfig storytellerConfig, FileUtils fileUtils) {
     this.config = storytellerConfig.getConfig();
     this.fileUtils = fileUtils;
+    allStories = new ArrayList<>();
+    currentStoryBuilder = Story.newBuilder();
   }
 
-  private String getSharedStoriesPath() {
-    return Storyteller.getSharedStoriesPath(config);
+  void startStory(String project) {
+    currentStoryBuilder
+        .setStartTimeMs(getCurrentTimestamp())
+        .setProject(project);
+    allStories.add(currentStoryBuilder.build());
+    saveStories();
   }
 
-  private String getUnsharedStoriesPath() {
-    return Storyteller.getUnsharedStoriesPath(config);
+  void updateStory(String project) {
```

Comment:
The way the code currently is, you can call `updateStory()` without ever calling `startStory()`.
How about not setting `currentStoryBuilder` in the constructor (leaving it null), initializing it in `startStory()`, and if it's not initialized in `updateStory()` or `endStory()`, throwing an exception?